calculate_trigger.py

```diff
@@ -52,9 +74,39 @@ def search_for_fault_location(
         goldenrun_tb_info.at[idtbinfo, "assembler"],
         goldenrun_tb_info.at[idtbinfo, "id"],
     )
+    return [idx, ins]
+
+
+def search_for_fault_location(
+    filter_lists,
+    trigger_position,
+    fault_address,
+    trigger_occurrences,
+    fault_lifespan,
+    goldenrun_tb_exec,
+    goldenrun_tb_info,
+):
+    logger.info(f"Search trigger to fault INSN at 0x{fault_address:08x}")
+    [idx, ins] = find_fault(
+        fault_address, goldenrun_tb_exec, goldenrun_tb_info, trigger_occurrences
+    )
+    if idx < 0:
+        return [-1, trigger_occurrences, fault_lifespan]
     trigger_not_in_same_tb = 0
+    lifespan_diff = trigger_position + fault_lifespan
     trigger_position = trigger_position * (-1)
     while trigger_position != 0:
+        if idx < 0:
+            new_lifespan = calculate_lifespan_from_start(
+                filter_lists,
+                fault_address,
+                goldenrun_tb_exec,
+                goldenrun_tb_info,
+                trigger_occurrences,
+            )
+            if fault_lifespan != 0:
```

Comment:
This should already be checked before we calculate the lifespan from start. In this case we can return early, because the lifespan does not have to be adjusted.